### DISCUSSION TITLE
Recommended way to load PNG into texture from include_bytes!() ?

### DISCUSSION THREAD

**Discussion by surban** [asny/three-d-asset#27] [Q&A]
Hi!

I am embedding a PNG file in my Rust executable using the `include_bytes!()` macro. Is there a recommended way to load it into a three-d texture?

Thanks.

**Comment by asny** [comment]
Sorry for the late reply. I should really make a simple function for doing that so it's easier to do, but it does only require 3 lines of code. You need to create RawAssets manually and insert the PNG bytes, then you can deserialize it as if you loaded the image. See the documentation of the insert method here: https://docs.rs/three-d-asset/latest/three_d_asset/io/struct.RawAssets.html#method.insert

**Comment by surban** [comment]
Thanks!

This is what I did, but it felt a bit unnatural to do so.

**Comment by asny** [comment]
Yeah, I know. It's because one 3D model file often consist of several files (geometry, material, textures etc.) so we need everything loaded before we can deserialize. There is probably a better design, but well, this was the best I could think of. I've added a `deserialize` function that can be used to deserialize a single file in https://github.com/asny/three-d-asset/commit/8c7097ca5dd9a8082ac65f37c28b55a2d87456f4 🙂

**Comment by surban** [comment]
Thanks for the function.

However, when embedding resources using `include_bytes!` it would be necessary to copy the data using `.to_vec()` before calling `deserialize`. Would it also be possible to provide a function that takes `&[u8]`, or at least `&'static [u8]`?
